File: branches/pre-scala-migration/src/java/org/codehaus/testdox/intellij/inspections/AddTestMethodQuickFix.java
package org.codehaus.testdox.intellij.inspections;

import com.intellij.codeInspection.LocalQuickFix;
import com.intellij.codeInspection.ProblemDescriptor;
import com.intellij.openapi.project.Project;

import org.codehaus.testdox.intellij.TestDoxProjectComponent;

class AddTestMethodQuickFix implements LocalQuickFix {

    public String getName() {
        return "Add Test";
    }

    public String getFamilyName() {
        return "TestDox Quick Fixes";
    }

    public void applyFix(Project project, ProblemDescriptor problemDescriptor) {
        TestDoxProjectComponent.getInstance(project).getController().addTest();
    }
}

File: trunk/src/main/scala/org/codehaus/testdox/intellij/inspections/AddTestMethodQuickFix.scala
package org.codehaus.testdox.intellij.inspections

import com.intellij.codeInspection.LocalQuickFix
import com.intellij.codeInspection.ProblemDescriptor
import com.intellij.openapi.project.Project

import org.codehaus.testdox.intellij.TestDoxProjectComponent

class AddTestMethodQuickFix extends LocalQuickFix {

  val getName = "Add Test"
  val getFamilyName = "TestDox Quick Fixes"

  def applyFix(project: Project, problemDescriptor: ProblemDescriptor) { TestDoxProjectComponent.getInstance(project).getController.addTest() }
}
